kola/tests/misc/tls.go

```diff
@@ -0,0 +1,34 @@
+package misc
+
+import (
+	"fmt"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+)
+
+var (
+	urlsToFetch = []string{
+		"https://www.example.com/",
+		"https://www.wikipedia.org/",
+		"https://start.fedoraproject.org/",
+	}
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:              TestTLSFetchURLs,
+		ClusterSize:      1,
+		Name:             "coreos.tls.fetch-urls",
+		ExcludePlatforms: []string{"qemu"}, // Networking outside cluster required
+	})
+}
+
+func TestTLSFetchURLs(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	for _, url := range urlsToFetch {
+		c.MustSSH(m, fmt.Sprintf("curl %s", url))
+		c.MustSSH(m, fmt.Sprintf("wget --delete-after %s", url))
```

Comment:
The only default timeout for `wget` is the read-timeout at 900 seconds, in the case of broken TLS this would most likely hang for inordinate amounts of time.

Let's maybe switch this to `wget -T 30 -t 2 --delete-after %s` so we have a maximum of 1 minute trying to fetch the URL (the default tries are also 10).

We could also consider adding `-q` to quiet the logs down (as they aren't particularly useful).

Reply:
Thanks - adding these in now.

Adding the `-q` to `wget` sounds good, as the test will still fail in the case of an error due to the non-zero exit code `wget` will give. Seems there is not a clean way to have `wget` log errors only - however `-nv` gives one line of output only for successes (when downloading only one web page at a time), and one line for failures if we err on the side of logging more. Could `-nv` be worth using instead of `-q`?

Reply:
Yeah `-nv` works. Just trying to reduce the general amount of log spam, having a single line works.